sympy/solvers/solveset.py

```diff
@@ -1963,6 +2014,9 @@ def add_type(lhs, rhs, symbol, domain):
 
         if lhs.is_Add:
             result = add_type(lhs, rhs, symbol, domain)
+        elif _is_lambert(lhs, symbol):
+            # try to get solutions in form of Lambert
+            result = _solve_as_lambert(lhs, rhs, symbol, domain)
```

Comment:
This doesn't seem like the right place to call this function. Why is this necessary?

Reply:
Actually in the documentation of `_transolve` you may see that solving of lambert is included, so I think that this part is correct.

Reply:
Why is the solver being called at the end here after all the solvers have been called?

Reply:
Yes, I think this is unnecessary. Actually, the code should be `lhs - rhs` instead of `lhs` .